lib/services/local/vnet_config.go

```diff
@@ -117,5 +118,18 @@ func validateVnetConfig(vnetConfig *vnet.VnetConfig) error {
 			return trace.BadParameter("ipv4_cidr_range must be valid IPv4")
 		}
 	}
+	for _, zone := range vnetConfig.GetSpec().GetCustomDnsZones() {
+		suffix := zone.GetSuffix()
+		if len(suffix) == 0 {
+			return trace.BadParameter("custom_dns_zone must have a non-empty suffix")
+		}
+		// Some very basic validation that zone suffixes can only contain lowercase letters, numbers, . or -.
+		// This is true for all valid hostnames.
+		for _, r := range suffix {
+			if !(unicode.IsLower(r) || unicode.IsDigit(r) || r == '.' || r == '-') {
+				return trace.BadParameter("custom_dns_zone.suffix can only contain letters, numbers, '-', and '.'")
+			}
```

Comment:
could split on "." and use `validation.IsDNS1035Label(name)` (or 1123) for each part here

Reply:
thanks, didn't know about that lib, makes it much easier to do proper validation